src/Adapter/Product/CommandHandler/DeleteProductHandler.php

```diff
@@ -31,11 +31,15 @@
 use PrestaShop\PrestaShop\Adapter\Product\Repository\ProductRepository;
 use PrestaShop\PrestaShop\Core\Domain\Product\Command\DeleteProductCommand;
 use PrestaShop\PrestaShop\Core\Domain\Product\CommandHandler\DeleteProductHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
 
 /**
  * Handles @see DeleteProductCommand using legacy object model
+ *
+ * @todo: this command along with BulkDeleteProductCommand needs to be refactored to fully support shop constraint.
+ *        Then the DeleteFromShopsCommand should be refactored to a SetProductShopsCommand.
```

Comment:
@jolelievre added this todo reminder to fix these commands in dedicated PR. It would be too much for current one.